### PR TITLE
Homework 9

### PR DESCRIPTION
**Pull request by PashaMisan**
Ansible 3

### CODE REVIEW

File: ansible/deploy.yml [hunk 1]
```diff
@@ -1,6 +1,14 @@
...
+  become: yes
+  roles:
+    - db
+
 - name: Deploy Flask app and configure systemd service
   hosts: app
   become: yes
+  roles:
+    - app
   tasks:
```
Comment (saaverdo) [inline ansible/deploy.yml:12]: It's OK in this case.
But combining roles and tasks can be confusing.
Better use pre-tasks or post-tasks with roles.

File: ansible/deploy.yml [hunk 2]
```diff
@@ -13,9 +21,14 @@
...
-      include_vars:
-        file: vars.yml
+    - name: Copy local script to server
+      ansible.builtin.copy:
+          src: ../packer/scripts/user_data/app.sh
+          dest: /home/ubuntu/script.sh
+          mode: '0755'
+
+    - name: Run script on server
+      command: /home/ubuntu/script.sh
```
Comment (saaverdo) [inline ansible/deploy.yml:31]: Btw, You can add a kind of idempotent behavior here with `creates:` option
[link to Ansible docs](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/command_module.html#parameter-creates)
example:
```
 - name: Run script on server
      command: /home/ubuntu/script.sh
      creates: /home/ubuntu/app.env
```
in this case if file specified in `creates:` option exists module will be skipped.
It's not necessary here, but can be useful in future.

**Review by saaverdo — APPROVED** [review]
Very good!
Documentation for roles and homework - special 👍
Approved)

File: ansible/roles/db/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+- name: Create flask_db database
+  community.mysql.mysql_db:
+    name: "{{ mysql_db }}"
+    state: present
+    login_unix_socket: /var/run/mysqld/mysqld.sock
+    login_user: root
+
+- name: Grant privileges on flask_db to admin
+  community.mysql.mysql_user:
```
Comment (saaverdo) [inline ansible/roles/db/tasks/main.yml:33]: Module `community.mysql.mysql_user`  used twice.
You can just create database first and after that create user with required privileges.

File: ansible/roles/db/tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  loop: "{{ prepare_host_packages }}"
+
+- name: Install pymysql python package
+  ansible.builtin.pip:
+    name: pymysql
+    state: present
+
+- name: Create MySQL user
+  community.mysql.mysql_user:
+    name: "{{ mysql_user }}"
```
Comment (saaverdo) [inline ansible/roles/db/tasks/main.yml:17]: This task can be easily removed.

**Review by saaverdo — APPROVED** [review]

File: ansible/deploy.yml (hunk 2 above)
Comment (PashaMisan) [reply ansible/deploy.yml:31]: Cool, didn't know about that

File: ansible/deploy.yml (hunk 1 above)
Comment (PashaMisan) [reply ansible/deploy.yml:12]: I thought it would be more compact, for the future I will divide it into pre-task and post-task, thank you

**Pull request merged**